src/runtime/compact.cpp

```diff
@@ -168,14 +168,14 @@ void object_compactor::insert_string(object * o) {
 
 bool object_compactor::insert_constructor(object * o) {
     std::vector<object_offset> & offsets = m_tmp;
-    offsets.clear();
     bool missing_children = false;
     unsigned num_objs     = lean_ctor_num_objs(o);
-    for (unsigned i = 0; i < num_objs; i++) {
+    offsets.resize(num_objs);
+    for (unsigned i = num_objs; i-- > 0;) {
```

Comment:
Could you please use the style we use in the rest of the code base?
```cpp
        unsigned i = num_objs;
        while (i > 0) {
            i--;
            // ....
```
Same for the other for-loop in the commit.